### ISSUE TITLE
Software and Hardware Requirements to run the models

### ISSUE THREAD

**Issue by Madhukar7559** [abhineet123/river_ice_segmentation#1]
Hello There, I am Undergrad Student, I would like to know what are the Software Requirements (Python Packages, OS, and its versions) and Hardware Requirements ?

**Comment by abhineet123** [comment]
This repo is a collection of heterogenous modules and each one has its own package requirements, e.g. those for image-segmentation-keras are mentioned in its [readme](https://github.com/abhineet123/river_ice_segmentation/blob/master/image-segmentation-keras/README.md).

Some of these modules might not include package info or might have outdated info in which case I would recommend running the commands in the cmd files and installing missing packages as needed.

We have tested it with python 3.6 on Ubuntu 14.04, 16.04 and 18.04 although it should be able to work with any python version >= 3.5 and on most Linux distros.

We were able to run all the experiments using a couple of Geforce 1080 Ti GPUs so anything comparable should be sufficient.

**Comment by Madhukar7559** [comment]
Thank you for the reply. Is it possible to share weights (.h5) of different models ?

**Comment by abhineet123** [comment]
Yes, I do have the weights but there are too many to upload all of them. Let me know which specific weights you need.

**Comment by Madhukar7559** [comment]
Thank you for the reply, I am so sorry for the delay, is it possible to share all the weights of vgg_segnet, vgg_unet, Xception 65 (.h5 files) ?
Thank you.

**Comment by abhineet123** [comment]
As I said in my previous post, each of these models have far too many weights for all of them to be shared. 
Let me know which specific variant of these models described in the paper you'd like.

**Comment by Madhukar7559** [comment]
I am sorry I really didn't get the sentence "specific variant of these models described in the paper." Is it possible to provide an example ?
Thank you.

**Comment by abhineet123** [comment]
Never mind, I meant that I trained many different versions for each of these models for ablation testing but you probably just want the base models trained on images 0-31 with 640x640 patches.

These are available here:
[segnet](https://drive.google.com/file/d/1hEpB_DGY4608fxJPGWci73jhQ0Bhqvl6/view?usp=sharing)
[unet](https://drive.google.com/file/d/1-zjaV2QHxaGaxq3j8SpQvmfg7jRVnLwb/view?usp=sharing)
[xception65](https://drive.google.com/file/d/1DkWB8QuTiPx6Hz6cEhyJiVgoZRBDIjl3/view?usp=sharing)
